io/plugins/eu.esdihumboldt.hale.io.jdbc.mssql.test/src/eu/esdihumboldt/hale/io/jdbc/mssql/test/MsSqlSchemaReader.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.io.jdbc.mssql.test;
+
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+import eu.esdihumboldt.hale.common.core.io.supplier.LocatableInputSupplier;
+import eu.esdihumboldt.hale.io.jdbc.JDBCSchemaReader;
+
+/**
+ * An extension of {@link JDBCSchemaReader} for MsSQL Server Database
+ * 
+ * @author Arun
+ */
+public class MsSqlSchemaReader extends JDBCSchemaReader {
```

Comment:
This is in a test. I have to provide it to support special character in URI. I did it before I add general mechanism to support  special character in connection string.